test/integration/routing.go

```diff
@@ -174,3 +193,43 @@ func (t *routing) verifyFaultInjection(src, dst, headerKey, headerVal string,
 	}
 	return nil
 }
+
+// verifyRedirect verifies if the http redirect was setup properly
+func (t *routing) verifyRedirect(src, dst, targetHost, targetPath, headerKey, headerVal string, respCode int) error {
+	url := fmt.Sprintf("http://%s/%s", dst, src)
+	glog.Infof("Making 1 request (%s) from %s...\n", url, src)
+
+	resp := t.clientRequest(src, url, 1, fmt.Sprintf("-key %s -val %s", headerKey, headerVal))
+
+	if len(resp.code) == 0 || resp.code[0] != fmt.Sprintf(respCode) {
+		return fmt.Errorf("redirect verification failed: "+
+			"response status code: %v, expected %v",
+			resp.code, respCode)
+	}
+
+	exp := regexp.MustCompile("(?i)Host=(.*)")
+	hosts := exp.FindAllStringSubmatch(resp.body, -1)
+	host := ""
+	if len(hosts) > 0 {
+		host = hosts[0][1]
+	}
+	if host != targetHost {
+		return fmt.Errorf("redirect verification failed: "+
+			"response body contains Host=%v, expected Host=%v",
+			host, targetHost)
+	}
+
+	exp = regexp.MustCompile("(?i)URL=(.*)")
+	paths := exp.FindAllStringSubmatch(resp.body, -1)
+	path := ""
+	if len(paths) > 1 {
+		path = paths[1][1]
+	}
+	if path != targetPath {
+		return fmt.Errorf("redirect verification failed: "+
+			"response body contains URL=%v, expected URL=%v",
+			paths[0], targetPath)
```

Comment:
meant to be...?
`path, targetPath)`

Could panic, because this is outside of the `len(paths) > 1` check